dist/src/main/java/io/zeebe/broker/StandaloneBroker.java

```diff
@@ -9,24 +9,52 @@
 
 import static java.lang.Runtime.getRuntime;
 
+import io.zeebe.EnvironmentHelper;
 import io.zeebe.broker.system.configuration.BrokerCfg;
+import io.zeebe.legacy.tomlconfig.LegacyConfigurationSupport;
+import io.zeebe.legacy.tomlconfig.LegacyConfigurationSupport.Scope;
 import io.zeebe.util.FileUtil;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.concurrent.CountDownLatch;
 import org.apache.logging.log4j.LogManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.core.env.Environment;
 
-public class StandaloneBroker {
-  private static final CountDownLatch WAITING_LATCH = new CountDownLatch(1);
-  private static String tempFolder;
+@SpringBootApplication
+public class StandaloneBroker implements CommandLineRunner {
+
+  @Autowired BrokerCfg configuration;
+  @Autowired Environment springEnvironment;
+
+  private final CountDownLatch waiting_latch = new CountDownLatch(1);
+  private String tempFolder;
 
   public static void main(final String[] args) throws Exception {
+    System.setProperty("spring.banner.location", "classpath:/assets/zeebe_broker_banner.txt");
```

Comment:
What's the different between putting this here or in the launcher? Apart from showing it when not starting this from the distribution launcher?

Reply:
Not sure what you mean by launcher.

This has to be set before calling `SpringApplication.run(...)`. So you can set it here or in `startup.sh`, which would be weird in my mind. Not sure which other place you have in mind.

Reply:
I referred to the shell/batch scripts we generate in the packaging phase, e.g. `bin/broker`, in which we set some system properties when running the jar. These options are set in `dist/pom.xml`

Reply:
Well, I wouldn't set it there. I think the banner should be internal to the application. What I could do is set it only if it is undefined.

Reply:
I have no strong preference, I suppose I'm just used to setting properties externally, but I also have no strong argument against it. Let's go with this until we have a reason not to :+1: